datafusion/common/src/scalar/mod.rs

```diff
@@ -3027,7 +3028,13 @@ impl ScalarValue {
             },
             ScalarValue::Utf8View(e) => match e {
                 Some(value) => {
-                    Arc::new(StringViewArray::from_iter_values(repeat_n(value, size)))
+                    let mut builder =
+                        StringViewBuilder::with_capacity(size).with_deduplicate_strings();
+                    for _ in 0..size {
```

Comment:
🤔  we should prob have some kind of `append_n` to remove this boilerplate, in some future for arrow-rs